light-rpc/src/cli.rs

```diff
@@ -0,0 +1,104 @@
+    fn default() -> Config {
+        Config {
+            rpc_addr: SocketAddr::from(([127, 0, 0, 1], 8899)),
+            json_rpc_url: ConfigInput::default().json_rpc_url,
+            websocket_url: ConfigInput::default().websocket_url,
+            subsription_port : SocketAddr::from(([127, 0, 0, 1], 8901)),
+        }
+    }
+}
+
+/// Defines and builds the CLI args for a run of the benchmark
+pub fn build_args<'a, 'b>(version: &'b str) -> App<'a, 'b> {
+    App::new("lite rpc")
+        .about("a lite version of solana rpc to send and confirm transactions")
+        .version(version)
+        .arg(
+            Arg::with_name("json_rpc_url")
+                .short("u")
+                .long("url")
+                .value_name("URL_OR_MONIKER")
+                .takes_value(true)
+                .global(true)
+                .validator(is_url_or_moniker)
+                .help(
+                    "URL for Solana's JSON RPC or moniker (or their first letter): \
+                     [mainnet-beta, testnet, devnet, localhost]",
+                ),
+        )
+        .arg(
+            Arg::with_name("websocket_url")
+                .long("ws")
+                .value_name("URL")
+                .takes_value(true)
+                .global(true)
+                .validator(is_url)
+                .help("WebSocket URL for the solana cluster"),
+        )
+        .arg(
+            Arg::with_name("port")
+                .long("port")
+                .short("p")
+                .takes_value(true)
+                .global(true)
+                .help("Port on which which lite rpc will listen to rpc requests"),
+        )
+        .arg(
+            Arg::with_name("subsription_port")
+                .long("sub_port")
+                .short("sp")
+                .takes_value(true)
+                .global(true)
+                .help("subsription port on which which lite rpc will use to create subscriptions")
+        )
+}
+
+pub fn extract_args(matches: &ArgMatches) -> Config {
+    let mut args = Config::default();
+
+    let config = if let Some(config_file) = matches.value_of("config_file") {
+        solana_cli_config::Config::load(config_file).unwrap_or_default()
+    } else {
+        solana_cli_config::Config::default()
+    };
+    let (_, json_rpc_url) = ConfigInput::compute_json_rpc_url_setting(
+        matches.value_of("json_rpc_url").unwrap_or(""),
+        &config.json_rpc_url,
+    );
+    args.json_rpc_url = json_rpc_url;
+
+    let (_, websocket_url) = ConfigInput::compute_websocket_url_setting(
+        matches.value_of("websocket_url").unwrap_or(""),
+        &config.websocket_url,
+        matches.value_of("json_rpc_url").unwrap_or(""),
+        &config.json_rpc_url,
+    );
+    args.websocket_url = websocket_url;
+    if let Some(port) = matches.value_of("port") {
+        let port: u16 = port.parse().expect("can't parse port");
+        args.rpc_addr = SocketAddr::from(([127, 0, 0, 1], port));
```

Comment:
guess that does not work on a _real_ network - does it?

Reply:
No this port will be used for server to bind locally on given port.